lib/std/os/linux.zig

```diff
@@ -154,8 +154,22 @@ pub usingnamespace @import("linux/io_uring.zig");
 /// Set by startup code, used by `getauxval`.
 pub var elf_aux_maybe: ?[*]std.elf.Auxv = null;
 
-/// See `std.elf` for the constants.
-pub fn getauxval(index: usize) usize {
+pub usingnamespace if (switch (builtin.zig_backend) {
+    // Calling extern functions is not yet supported with these backends
+    .stage2_x86_64, .stage2_aarch64, .stage2_arm, .stage2_riscv64, .stage2_sparc64 => false,
+    else => !builtin.link_libc,
+}) struct {
+    /// See `std.elf` for the constants.
+    /// This matches the libc getauxval function.
+    pub extern fn getauxval(index: usize) usize;
+    comptime {
+        @export(getauxvalImpl, .{ .name = "getauxval", .linkage = .Weak });
```

Comment:
As I understand, we are exporting it as weak because we expect it to be overriden by a different version in an exe/dso that has `getauxval` hooked up into startup routine? What if we're building an exe and link with a dso? Which version takes precedence then? Do I even understand the problem correctly here?

Reply:
> As I understand, we are exporting it as weak because we expect it to be overriden by a different version in an exe/dso that has getauxval hooked up into startup routine?

Yes, exactly. The intention is that version used is they one that references the `elf_aux_maybe` initialized by the startup code.

> What if we're building an exe and link with a dso? Which version takes precedence then? Do I even understand the problem correctly here?

My assumption had been that the version in the exe would take precedence, but if this is not true in all cases then this solution won't be adequate.

Reply:
Gotcha thanks!